tiled/iterviews.py

```diff
@@ -0,0 +1,205 @@
+"""
+Iterables for KeysView, ValuesView, ItemsView that are sliceable.
+"""
+
+
+class IterViewBase:
+
+    __slots__ = ("_get_length",)
+
+    def __init__(self, get_length):
+        self._get_length = get_length
+
+    def __repr__(self):
+        return f"<{type(self).__name__}>"
+
+    # Convenience aliases
+
+    def first(self):
+        return self[0]
+
+    def last(self):
+        return self[-1]
+
+    def head(self, n=5):
+        return self[:n]
+
+    def tail(self, n=5):
+        return list(reversed(self[-1 : -(n + 1) : -1]))  # noqa: E203
```

Comment:
I like the idea of using methods like `first()`, `last()`, `head()` and `tail()`. It might be silly but I think we should check for cases smaller than the default value. After dealing with some cases where people saved datasets with no data or just one sample, I always try and check for size before indexing. For example, what if the dataset is len(self)<5 and the method uses the default value for n (n=5)?

Reply:
Good question. We are protected there by Python's slicing behavior. If you slice "the first 100" in a list of length 3, you get 3 items back, not an error. The slicing is interpreted to mean _up to_ the first 100. This usually the desired behavior.

```py
>>> ['a', 'b', 'c'][:100]
['a', 'b', 'c']
```

On the other hand if you ask for the 100th item specifically, not a slice by an integer lookup, then you get an `IndexError`.

```py
>>> ['a', 'b', 'c'][100]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: list index out of range
```

I added some tests just now ([91b3f9f](https://github.com/bluesky/tiled/pull/216/commits/91b3f9fcbd24ed1625d7a5109527b19923d3c707)) to verify that calling `.head()` on an empty tree returns an empty list, whereas calling `.first()` raises an `IndexError`.